### ISSUE TITLE
Hardcoded llvmorg-13.0.1/clang+llvm-13.0.1-x86_64-linux-gnu-ubuntu-18.04.tar.xz

### ISSUE THREAD

**Issue by catalogoanatom** [plv8/plv8js#536]
In v 3.1.6 Makefile.linux has hardcoded link to llvm ubuntu 18.04.
```
 ARCH := $(shell uname -m)
ifeq ($(ARCH),aarch64)
        LLVM := https://github.com/llvm/llvm-project/releases/download/llvmorg-13.0.1/clang+llvm-13.0.1-aarch64-linux-gnu.tar.xz
        LLVM_BINPATH := $(shell pwd)/build/clang+llvm-13.0.1-aarch64-linux-gnu
        ARCH := arm64
endif
ifeq ($(ARCH),x86_64)
        LLVM := https://github.com/llvm/llvm-project/releases/download/llvmorg-13.0.1/clang+llvm-13.0.1-x86_64-linux-gnu-ubuntu-18.04.tar.xz
        LLVM_BINPATH := $(shell pwd)/build/clang+llvm-13.0.1-x86_64-linux-gnu-ubuntu-18.04
        ARCH := x64
endif
```
Will it work properly in case of build for Ubuntu 22.04? Last currently available llvm is :
https://github.com/llvm/llvm-project/releases/download/llvmorg-16.0.4/clang+llvm-16.0.4-x86_64-linux-gnu-ubuntu-22.04.tar.xz

**Comment by JerrySievert** [comment]
unfortunately, at this time llvm 16 and postgres are incompatible with each other.

when this is fixed on the postgres side, I can look at a newer version for plv8.  essentially, the JIT building process changed in llvm 16, and the postgres maintainers are working on migrating.

**Comment by catalogoanatom** [comment]
The question is about compability clang+llvm-13.0.1-x86_64-linux-gnu-**ubuntu-18.04** with Ubuntu 22.04.  I have successfully built it 3.1.6, but I have a doubt how it will work in future.

**Comment by JerrySievert** [comment]
there are a couple of issues at play here:

1. due to build requirements of v8, we are very limited to what versions of what compilers work with exactly which versions of v8 - this is a problem, as it doesn't allow for us to choose build tools, so it is very hard to validate what will work in exactly which linux installation and what will not work
2. postgres uses llvm's JIT system for its own JIT implementation, and in llvm 16, the method that postgres uses was removed.  since part of the extension build process is postgres building the JIT aspects for its own use, llvm 16 simply will not work correctly with postgres or extensions - you can see some discussion of that on the [official postgres bugs mailing list](https://www.postgresql.org/message-id/7AA0E02C-F7FB-4515-ACC5-306B791D2E61@yesql.se)

so, even if the first issue were a simple matrix and checkbox, the second issue is a complete blocker until the core postgres team can address the deprecation of typed pointers in llvm 16.

v8 requiring a sysroot of ubuntu 18.04 during its compilation phase is not an issue, but using the only sysroot available for 22.04 (llvm 16) is an issue.

will it break on 22.04 with llvm 13/18.04?  the likelihood is extremely low, since it isn't building a kernel, and is just using the 18.04 sysroot.

will it break with llvm 16?  absolutely 100% and until postgres can build with llvm 16, it's a non-starter to even try to validate whether v8 will build with llvm 16 (which, given v8's prior build pains is a coin flip at this point at best).

does that help answer your questions?

**Comment by catalogoanatom** [comment]
thanks
